src/shared/components/header/header.component.ts

```diff
@@ -1,15 +1,56 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, DestroyRef, OnInit } from '@angular/core';
 import { Dialog } from '@angular/cdk/dialog';
 import { TodoFormComponent } from '../todo-form/todo-form.component';
+import { SortFacadeService } from '../../services/sort/sort-facade.service';
+import { FormControl } from '@angular/forms';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { distinctUntilChanged } from 'rxjs';
+import { SortTerm } from '../../enums';
+import { IOption } from '../../interfaces';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class HeaderComponent {
-  constructor(public dialog: Dialog) {}
+export class HeaderComponent implements OnInit {
+  options: IOption[] = [
+    {
+      id: SortTerm.date,
+      value: `${ this.buildTranslationKey(SortTerm.date) }`,
+    },
+    {
+      id: SortTerm.status,
+      value: `${ this.buildTranslationKey(SortTerm.status) }`,
+    },
+    {
+      id: SortTerm.name,
+      value: `${ this.buildTranslationKey(SortTerm.name) }`,
+    },
+  ];
+
+  iconSrc = 'assets/icons/sort.svg';
```

Comment:
unused variable

Reply:
it is not unused, I use it to display the icon inside sort drop down

Reply:
Please check the solution in dropdown.component.html, I am not sure about it.